lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -3,109 +3,137 @@
 import { CacheHelper } from "./utils/CacheHelper";
-import { CacheRecord } from "./entities/CacheRecord";
 import { StringUtils } from "../utils/StringUtils";
 import { IdTokenEntity } from "./entities/IdTokenEntity";
-import { AccessTokenEntity } from "./entities/AccessTokenEntity";
 import { RefreshTokenEntity } from "./entities/RefreshTokenEntity";
-import { ScopeSet } from "../request/ScopeSet";
+import { AuthError } from "../error/AuthError";
+import { ICacheManager } from "./interface/ICacheManager";
+import { IAccount } from "../account/IAccount";
+import { ClientAuthError } from "../error/ClientAuthError";
 
-export class UnifiedCacheManager implements ICacheManager {
-    // Storage interface
-    private cacheStorage: ICacheStorage;
-    private inMemory: boolean;
-    private clientId: string;
+/**
+ * Interface class which implement cache storage functions used by MSAL to perform validity checks, and store tokens.
+ */
+export abstract class CacheManager implements ICacheManager {
 
-    constructor(cacheImpl: ICacheStorage, clientId: string, storeInMemory: boolean) {
-        this.cacheStorage = cacheImpl;
-        this.clientId = clientId;
-        this.inMemory = storeInMemory;
-    }
+    /**
+     * Function to set item in cache.
+     * @param key
+     * @param value
+     */
+    abstract setItem(key: string, value: string | object, type?: string): void;
+
+    /**
+     * Function which retrieves item from cache.
+     * @param key
+     */
+    abstract getItem(key: string, type?: string): string | object;
 
     /**
-     * Initialize in memory cache from an exisiting cache vault
-     * @param cache
+     * Function to remove an item from cache given its key.
+     * @param key
      */
-    generateInMemoryCache(cache: string): InMemoryCache {
-        return Deserializer.deserializeAllCache(
-            Deserializer.deserializeJSONBlob(cache)
-        );
-    }
+    abstract removeItem(key: string, type?: string): boolean;
+
+    /**
+     * Function which returns boolean whether cache contains a specific key.
+     * @param key
+     */
+    abstract containsKey(key: string, type?: string): boolean;
 
     /**
-     * retrieves the final JSON
-     * @param inMemoryCache
+     * Function which retrieves all current keys from the cache.
      */
-    generateJsonCache(inMemoryCache: InMemoryCache): JsonCache {
-        return Serializer.serializeAllCache(inMemoryCache);
-    }
+    abstract getKeys(): string[];
+
+    /**
+     * Function which clears cache.
+     */
+    abstract clear(): void;
 
     /**
-     * Returns all accounts in memory
+     * Returns all accounts in cache
      */
-    getAllAccounts(): AccountCache {
-        return this.getAccountsFilteredBy();
+    getAllAccounts(): IAccount[] {
+        const currentAccounts: AccountCache = this.getAccountsFilteredBy();
+        const accountValues: AccountEntity[] = Object.values(currentAccounts);
```

Comment:
Object.values won't work in IE11 afaik. I think there's already a lot of incompatibility in 2.0 so this won't be the thing that breaks it, but its more to clean up later if/when we do work on compatibility.

Reply:
good call. @pkanher617 can you please note this to track when we do a recon on msal-common?